test/test_functions/test_multi_objective.py

```diff
@@ -75,6 +75,8 @@ def test_base_mo_problem(self):
                     self.assertTrue(torch.equal(f_X, expected_f_X))
                 with self.assertRaises(NotImplementedError):
                     f.gen_pareto_front(1)
+            with self.assertRaises(InputDataError):
```

Comment:
Good practice to use `assertRaisesRegex` to make sure that the right error message is being raised (and that this doesn't come from some other similar error)

```suggestion
            with self.assertRaisesRegex(InputDataError, "must match the number of objectives"):
```